src/share/classes/com/sun/javatest/regtest/config/RegressionEnvironment.java

```diff
@@ -46,5 +48,27 @@ public TestEnvironment copy() {
         return new RegressionEnvironment(this);
     }
 
+    /**
+     * {@return the hostname of the host in which this test harness is being run}
+     */
+    public final String getHostName() {
+        return CachedCanonicalHostName.HOST_NAME;
+    }
+
     public final RegressionParameters params;
+
+
+    private static final class CachedCanonicalHostName {
+        private static final String HOST_NAME;
+
+        static {
+            String hostname;
+            try {
+                hostname = InetAddress.getLocalHost().getCanonicalHostName();
+            } catch (UnknownHostException e) {
+                hostname = "127.0.0.1";
```

Comment:
I'm not sure "127.0.0.1" is a canonical hostname ;-)

Reply:
`InetAddress.getCanonicalHostName()`'s API allows it to return a textual representation of IP address if it isn't able to determine the fully qualified domain name. So I think naming this class `CachedCanonicalHostName` and yet continuing to return an IP address as a fallback is OK.